src/reducer/resource/list/ids.js

```diff
@@ -1,12 +1,19 @@
-import { CRUD_GET_LIST_SUCCESS } from '../../../actions/dataActions';
+import { CRUD_GET_LIST_SUCCESS, CRUD_DELETE_SUCCESS } from '../../../actions/dataActions';
 
-export default (resource) => (previousState = [], { type, payload, meta }) => {
+export default resource => (previousState = [], { type, payload, requestPayload, meta }) => {
     if (!meta || meta.resource !== resource) {
         return previousState;
     }
     switch (type) {
     case CRUD_GET_LIST_SUCCESS:
         return payload.data.map(record => record.id);
+    case CRUD_DELETE_SUCCESS: {
```

Comment:
This modifies the list before it's loaded. So even without a good connectivity, a deleted element disappears from the list.